ciao-image/datastore/store.go

```diff
@@ -178,11 +178,14 @@ func (s *ImageStore) UploadImage(ID string, body io.Reader) error {
 	if s.rawDs != nil {
 		err = s.rawDs.Write(ID, body)
 		if err != nil {
+			database.Logger.Errorf("Could not write image: %v", err)
+			// TODO shouldn't we return err here?
 			img.State = Killed
 		}
 
 		img.Size, err = s.rawDs.GetImageSize(ID)
 		if err != nil {
+			database.Logger.Errorf("Could not get image size: %v", err)
 			img.State = Killed
 			return err
```

Comment:
I think this code should be 

```
if err != nil {
  database.Logger.Errorf("Could not get image size: %v", err)
  img.State = Killed
  _ s.rawDS.Delete(ID)
}
```

@obedmr, @rbradford Could you comment on this?